compiler/src/dotty/tools/dotc/transform/init/Summary.scala

```diff
@@ -12,65 +13,80 @@ import config.Printers.init
 
 import Potentials._, Effects._, Util._
 
-object Summary {
-  type Summary = (Potentials, Effects)
-  val empty: Summary = (Potentials.empty, Effects.empty)
-
-  /** Summary of class.
-   *
-   *  It makes ObjectPart construction easier with already established raw outer for parents.
-   */
-  case class ClassSummary(currentClass: ClassSymbol, parentOuter: Map[ClassSymbol, Potentials]) {
-    private val summaryCache: mutable.Map[Symbol, Summary] = mutable.Map.empty
-
-    def cacheFor(member: Symbol, summary: Summary)(using Context): Unit = {
-      traceIndented("cache for " + member.show + ", summary = " + Summary.show(summary), init)
-      assert(member.owner == currentClass, "owner = " + member.owner.show + ", current = " + currentClass.show)
-      summaryCache(member) = summary
-    }
+case class Summary(pots: Potentials, effs: Effects) {
```

Comment:
What is the difference between `object` and `case class`?

Reply:
Here we use explicit class for `Summary`, instead of putting them in a tuple `(Potentials, Effects)`. The former is expected to be more maintainable.